backend/src/main/kotlin/app/ehrenamtskarte/backend/auth/webservice/schema/SignInMutationService.kt

```diff
@@ -6,17 +6,28 @@ import app.ehrenamtskarte.backend.auth.webservice.schema.types.Administrator
 import app.ehrenamtskarte.backend.auth.webservice.schema.types.AuthData
 import app.ehrenamtskarte.backend.auth.webservice.schema.types.Role
 import app.ehrenamtskarte.backend.auth.webservice.schema.types.SignInPayload
+import app.ehrenamtskarte.backend.common.webservice.GraphQLContext
 import com.expediagroup.graphql.generator.annotations.GraphQLDescription
 import com.expediagroup.graphql.generator.exceptions.GraphQLKotlinException
 import org.jetbrains.exposed.sql.transactions.transaction
+import graphql.schema.DataFetchingEnvironment
+import org.slf4j.LoggerFactory
 
 @Suppress("unused")
 class SignInMutationService {
     @GraphQLDescription("Signs in an administrator")
-    fun signIn(project: String, authData: AuthData): SignInPayload {
+    fun signIn(project: String, authData: AuthData, dfe: DataFetchingEnvironment): SignInPayload {
+        val logger = LoggerFactory.getLogger(SignInMutationService::class.java)
+
         val administratorEntity = transaction {
             AdministratorsRepository.findByAuthData(project, authData.email, authData.password)
-        } ?: throw GraphQLKotlinException("Invalid credentials")
+        } 
+        if (administratorEntity == null) {
+            val context = dfe.getContext<GraphQLContext>()
+            logger.info("${context.remoteIp} ${authData.email} failed to log in")
```

Comment:
@svenseeberg does that work for you to block users with repreated failed logins?